### PR TITLE
[ASTextNode] Factor in constrained size for final size calculation of ASTextNode

### PR DESCRIPTION
**Pull request by maicki**
cc @Adlai-Holler @garrettmoon

### CODE REVIEW

File: AsyncDisplayKit/ASTextNode.mm [hunk 1]
```diff
@@ -397,11 +401,12 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
...
   }
   
-  //add textContainerInset
+  // Add the constrained size back textContainerInset
   size.width += (_textContainerInset.left + _textContainerInset.right);
   size.height += (_textContainerInset.top + _textContainerInset.bottom);
   
-  return size;
+  return CGSizeMake(fminf(size.width, originalConstrainedSize.width),
+                    fminf(size.height, originalConstrainedSize.height));
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASTextNode.mm]: Use std::min or whatever here, so that it's correct on 64-bit

Comment (maicki) [inline AsyncDisplayKit/ASTextNode.mm]: Done

**Comment by Adlai-Holler** [comment]
Cool!

File: AsyncDisplayKit/ASTextNode.mm [hunk 2]
```diff
@@ -397,11 +401,12 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
     }
   }
   
-  //add textContainerInset
+  // Add the constrained size back textContainerInset
```
Comment (garrettmoon) [inline AsyncDisplayKit/ASTextNode.mm]: This is actually adding back in the textContainerInset to the returned text size.

**Pull request merged**